batch/batch/batch.py

```diff
@@ -420,13 +422,14 @@ def __init__(self, batch_id, job_id, attributes, callback, userdata, user, alway
         self.userdata = userdata
         self.user = user
         self.exit_codes = exit_codes
-        self.duration = duration
         self.log_uris = log_uris
         self.pod_statuses = pod_statuses
+        self.duration = duration
+        self.token = token
 
-        self._pvc_name = pvc_name
+        self._pod_name = f'batch-{batch_id}-job-{job_id}-pod-{token}'
+        self._pvc_name = f'batch-{batch_id}-job-{job_id}-pvc-{token}'
```

Comment:
we don't need to include `-pod-` or `-pvc-` if we don't want to, they're different resources so they're allowed to have the same name.